include/boost/url/impl/decoded_view.hpp

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_IMPL_DECODED_VIEW_HPP
+#define BOOST_URL_IMPL_DECODED_VIEW_HPP
+
+namespace boost {
+namespace urls {
+
+class decoded_view::iterator
+{
+    char const* begin_{nullptr};
+    char const* pos_{nullptr};
+    char const* end_{nullptr};
+    bool plus_to_space_{true};
+
+    friend decoded_view;
+
+    iterator(
+        char const* str,
+        size_type n,
+        bool plus_to_space) noexcept
+        : begin_(str)
+        , pos_(str)
+        , end_(str + n)
+        , plus_to_space_(plus_to_space)
+    {}
+
+    // end ctor
+    iterator(
+        char const* str,
+        size_type n,
+        bool plus_to_space,
+        int) noexcept
+        : begin_(str)
+        , pos_(str + n)
+        , end_(str + n)
+        , plus_to_space_(plus_to_space)
+    {}
+
+public:
+    using value_type = char;
+    using reference = char;
+    using pointer = void const*;
+    using const_reference = char;
+    using size_type = std::size_t;
+    using difference_type = std::ptrdiff_t;
+    using iterator_category =
+        std::bidirectional_iterator_tag;
+
+    iterator() = default;
+
+    iterator(iterator const&) = default;
+
+    iterator&
+    operator=(iterator const&) = default;
+
+    reference
+    operator*() const noexcept
+    {
+        char c{'\0'};
+        pct_decode_opts opt;
+        opt.plus_to_space = plus_to_space_;
+        pct_decode_unchecked(
+            &c, &c+1, string_view(pos_, end_), opt);
+        return c;
+    }
+
+    iterator&
+    operator++() noexcept
+    {
+        BOOST_ASSERT(pos_ != nullptr);
+        if (*pos_ != '%')
+            ++pos_;
+        else
+            pos_ += 3;
+        return *this;
+    }
+
+    iterator&
+    operator--() noexcept
+    {
+        BOOST_ASSERT(pos_ != begin_);
+        if (pos_ - begin_ < 3 ||
+            pos_[-3] != '%')
+            --pos_;
```

Comment:
I indent the last line of a multiline condition when the body of the `if` has a single statement with no braces:
```
        if (pos_ - begin_ < 3 ||
                pos_[-3] != '%')
            --pos_;
```

But that's just style

Reply:
I fixed it, but I'm not sure this is going to be easy to remember next time. I'll probably have to warn me again.